### ISSUE TITLE
CORE: Having a demo mode

### ISSUE THREAD

**Issue by cwang** [docqai/docq#20]
**Is your feature request related to a problem? Please describe.**

A few problems with a shared demo app like Docq:
- Data needs to be wiped from time to time
- Some demo data can be helpful, if pre-populated

**Describe the solution you'd like**

- Implementing a demo mode turned on by a flag (env var). 
- add a permanent banner, visible on every screen, that clearly indicates that this is a shared environment for demo purposes and not to upload sensitive data. In red colour. 

**Describe alternatives you've considered**

N/A

**Additional context**

Mostly about data preparation and deletion under the demo mode. Welcome other ideas to be built in.

**Comment by janaka** [comment]
Add what data to pre-populate and how that will work, which data etc. (@cwang)

**Comment by cwang** [comment]
For demo mode
- It's triggered by https://github.com/docqai/docq/blob/305e414b078831008f34c4865104291de82ece65/source/docq/config.py#L6
- It should populate data for
  - A shared internal space called "Demo Shared Space"
  - A shared external space called "Demo Public Space"
  - A few documents in both the personal space and the newly created shared spaces
  - A few non-admin users

**Comment by cwang** [comment]
@janaka think of some scenarios where we can using these demo documents to build a story about how an org would use these features? I think this is the true purpose of having demo data.

**Comment by janaka** [comment]
> @janaka think of some scenarios where we can using these demo documents to build a story about how an org would use these features? I think this is the true purpose of having demo data. 

Sure thing.

This what you were going to brain dump use case ideas you had for a blog post right?

**Comment by janaka** [comment]
Let's put this on-hold until we have orgs i.e. multi tenant and then reevaluate if we really need this.

**Comment by janaka** [comment]
Idea: Once we have orgs ( #44 ) we can enable throw away orgs in demo mode. 

- demo mode enabled and defaults as per [above](https://github.com/docqai/docq/issues/20#issuecomment-1619926543)
- perma banner fix to the top displayed very clearly indicating demo mode, don't share sensitive data, will be auto wipped after X  time. Exact copy TBD.
- user is prompted to end an email address only instead of login.
  - this auto creates an org based on the email address prefix
  - and a user based on the email address (as admin)
  - add default data
  - user sliding session auth from fix for #37 should work here. If an email address with an expired session is entered then reset (wipe data, recreate org, user etc.)
- have all the demo logic executed from a single place so it's dead simple to reason. We don't want `if DEMO` logic spaghetti all over the code base.


The above is with prospect self-service demos in mind. For sales demos that require per-prospect personalisation, would be setup in a different instance of Docq that's running in production mode.


Alt idea: rather than have a demo mode we go all the way to enable self-serve sign up (still depends on #44) which can serve all demo use cases both self serve and field sales assist. This is a more leveraged investment. But a little bit more work.

**Comment by janaka** [comment]
No longer needed
